discovery_engine_core/core/src/engine.rs

```diff
@@ -76,6 +81,40 @@ pub struct Config {
     kpe_classifier: String,
 }
 
+#[allow(dead_code)]
+/// Creates a discovery engine with [`xayn_ai::ranker::Ranker`] as a ranker.
+fn create_engine(
```

Comment:
i'd suggest some renaming of the constructors of `Engine` since we have more and more of them to better reflect their intention:
- `create_engine()` -> `from_config()` because this actually uses the config for construction
- `new()` -> `from_state()` because this uses the state for construction
- `from_config()` -> `new()` (or maybe `default()`) because this uses none of the above and creates new empty stack data (without using the config)

wdyt?